### PR TITLE
Enhancement: Add ES and UMD builds

### PR DESCRIPTION
**Pull request by aweary**
* Adds an ES build with ES6 modules
* Adds a UMD build using rollup

With the UMD build both `prop-types` and `react` are expected to be available on the global object before Freactal is loading. Since both of those packages have UMD builds, this should be fine.

-----

The maintainers of this repo require that all pull request submitters adhere to the following:

- [x] I have read and will comply with the [contribution guidelines](https://github.com/FormidableLabs/freactal/blob/master/CONTRIBUTE.md).
 _(required)_
- [x] I have read and will comply with the [code of conduct](https://github.com/FormidableLabs/freactal/blob/master/CONTRIBUTE.md).
 _(required)_
- [x] All related documentation has been updated to reflect the changes made. _(required)_
- [x] My commit messages are cleaned up and ready to merge. _(required)_

The maintainers of this repository require you to select the semantic version type that
the changes in this pull request represent.  Please select one of the following:
- [ ]  major
- [ ]  minor
- [x]  patch
- [ ]  documentation only

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  output: {
+    file: `./umd/${name}`,
+    directory: "umd",
+    format: "umd"
+  },
+  name: "Freactal",
+  external: ["react", "prop-types"],
+  globals: {
+    "react": "React",
+    "prop-types": "PropTypes"
```
Comment (ryan-roemer) [inline rollup.config.js]: While we're at it -- should we move `prop-types` from `dependencies` to `peerDependencies`? That's typically how I see most projects doing it these days...

/cc @divmain

**Comment by aweary** [comment]
@ryan-roemer I've checked all the required boxes and moved `prop-types` to the peer dependencies.

File: rollup.config.js (hunk 1 above)
Comment (ryan-roemer) [reply rollup.config.js]: Turns out documentation says the other way (actual dep now...) I'll switch it back.

**Comment by ryan-roemer** [comment]
Closing in favor of https://github.com/FormidableLabs/freactal/pull/94 -- @aweary thanks for all the great work, I've just move almost entirely what you've got here to there :)

**Pull request closed without merging**